### ISSUE TITLE
Auto category axis label width or percent

### ISSUE THREAD

**Issue by Eldair** [amcharts/amcharts4#959]
Hi,

Is it possible to have auto truncating of category axis labels based on grid item width (column width in bar chart) or at least an option to set maxWidth as a percent relative to aforementioned item?

Another possibility would be to have auto-hiding of labels without setting minGridDistance (it is in pixels as well)?

**Comment by martynasma** [comment]
Oops, just seeing this.

Challenge, of course, accepted :)

```
categoryAxis.renderer.labels.template.truncate = true;
categoryAxis.events.on("sizechanged", function(ev) {
  var axis = ev.target;
  var cellWidth = axis.pixelWidth / (axis.endIndex - axis.startIndex);
  axis.renderer.labels.template.maxWidth = cellWidth;
});
```

https://codepen.io/team/amcharts/pen/PLqEdO

**Comment by Eldair** [comment]
Trying it now (: Thanks!

**Comment by Eldair** [comment]
@martynasma is there a way to get the number of visible labels (to use instead of axis.endIndex - axis.startIndex) in cases some of the labels are hidden due to minGridDistance?

**Comment by martynasma** [comment]
Technically, we always have private property `_frequency` which can be used to calculate number of labels based on startIndex and endIndex.

I'm gonna expose it via readonly accessor `frequency`. But it will be only available in the next version. (at some point next week I presume)

Meanwhile, if you are using JS version you can light a candle for gods of OOP and just use `axis._frequency` :)

**Comment by martynasma** [comment]
Fixed in [4.2.0](https://github.com/amcharts/amcharts4/releases/tag/4.2.0).

## [4.2.0] - 2019-03-06

### Added
- Added read-only property of `CategoryAxis`: `frequency`. Indicates every X label is shown.
- Added JSON example of Candle stick chart (json/candlestick-series).
- New `Export` property: `extraSprites`. May contain a list of references to other `Sprites` (e.g. external legend or even other chart) to attach to exported image. [More info](https://www.amcharts.com/docs/v4/concepts/exporting/#Including_external_elements).
- Experimental responsive features and default rules are now active. [More info](https://www.amcharts.com/docs/v4/concepts/responsive/).

### Changed
- Accessibility: Default role of the Legend's item containers was changed to "switch".

### Fixed
- Accessibility: Legend's containers when toggling items were setting value of `aria-checked` incorrectly.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.

**Comment by Eldair** [comment]
Amazing (: that did the trick! Thank you very much
